### PR TITLE
Update Redoc for more OpenAPI/Swagger Goodies

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR updates the version of Redoc used in the docs to render Swagger/OpenAPI schema. It is part of the overall API updating needed for https://github.com/hypothesis/product-backlog/issues/539

In addition, the `group-schema.json` schema file has been removed, and that schema moved into `hypothesis.yaml` (the main OpenAPI schema file).

While I'm actually quite a fan of JSONSchema, I feel like its use here (versus OpenAPI/Swagger markup) is a little forced for a few reasons:

* It's an extra layer of abstraction in an already-complex doc-build process
* The JSON schemas here are not being used anywhere else (e.g. for request validation)...the JSON schema files at present are _solely_ to populate the docs, AFAICT
* There are a slew of OpenAPI features that we can't get at from JSONSchema objects—they have to be "real" OpenAPI schema. An example is the ability to mark a service, parameter or schema as deprecated.
* JSON isn't bad to work with, but given the choice, I find YAML even easier to compose and scan

If the above isn't objectionable, I may continue to fold in schema from JSONSchema files into the docs.

However, if there is a long-term plan for _why_ we've got external JSON schema for documetation, let me know and I'll abort.

Updating to new Redoc and a bit of light housekeeping in this PR snazzes up the docs a touch and makes them easier to navigate. Here are some before-and-afters!

**BEFORE: Lame navigation**

image

**AFTER: Somewhat more sane URIs**

image

**BEFORE: No real ordering or grouping for endpoints**

image

**AFTER: Endpoints are grouped by resource**

image

(unfurled:)

image

**BEFORE: No support for deprecated items**

image

**AFTER: built-in deprecation support/styling**

image

### CODE REVIEW

**Comment by robertknight** [comment]
> However, if there is a long-term plan for why we've got external JSON schema for documetation, let me know and I'll abort.

I don't personally know of reasons beyond "this is what redoc uses" and "we use JSON schema for other things so it is familiar". Some Slack/Git/GH history searches for OpenAPI / Swagger may turn up more details. 

My personal criteria for whatever tools we end up using would be something that allows us to provide:

- Good documentation which is easy to maintain
- Validation of request structures
- Automated verification that request/response formats match the documentation
- (Nice to have) An online REPL for testing out API queries
- (Nice to have) Automated generation of code for API clients

I'm not familiar enough with the current ecosystem to make concrete recommendations.

File: docs/_extra/api-reference/hypothesis.yaml [hunk 1]
```diff
@@ -391,7 +425,53 @@ definitions:
     items:
       $ref: '#/definitions/Group'
   Group:
-    $ref: './schemas/group-schema.json'
+    type: object
```
Comment (robertknight) [inline docs/_extra/api-reference/hypothesis.yaml]: If we use yaml is it still possible to extract the schemas for data types into a separate file, in order to keep `hypothesis.yaml` to a size where it is still easy to skim through?

Comment (lyzadanger) [reply docs/_extra/api-reference/hypothesis.yaml]: Absolutely something on my list to figure out—this PR is not comprehensive (to say the least) in terms of moving toward an ideal organization of stuff. I agree that the current YAML file could get unwieldy.

**Comment by lyzadanger** [comment]
@robertknight 

> I would be interested in finding automated ways to ensure that documentation and actual request/response structure are all kept in sync, and detailed enough to actually use.

ME. TOO. :) Thanks for the review!

**Pull request merged**